src/software/thunderscope/thunderscope.py

```diff
@@ -247,34 +498,168 @@ def close(self):
+        ) as full_system:
+            tscope.load_saved_layout(
+                args.layout, load_blue=args.run_blue, load_yellow=args.run_yellow
+            )
+            tscope.show()
+
+    ###########################################################################
+    #           Blue AI vs Yellow AI + Simulator + Gamecontroller             #
+    ###########################################################################
+    #
+    # Run two AIs against each other with the er-force simulator. We also run
+    # the gamecontroller which can be accessed from http://localhost:8081
+    #
+    # The async sim ticket ticks the simulator at a fixed rate.
     else:
-        thunderscope = Thunderscope()
-        thunderscope.configure_default_layout()
-        thunderscope.show()
+
+        def __async_sim_ticker(tick_rate_ms):
+            """Setup the world and tick simulation forever
+
+            :param tick_rate_ms: The tick rate of the simulation
+
+            """
+            world_state = tbots_protobuf.create_world_state(
+                [geom.Point(3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)],
+                [geom.Point(-3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)],
+                ball_location=geom.Point(0, 0),
+                ball_velocity=geom.Vector(0, 0),
+            )
+            tscope.simulator_proto_unix_io.send_proto(WorldState, world_state)
+
+            # Tick Simulation
+            while True:
+                tick = SimulatorTick(milliseconds=tick_rate_ms)
+                tscope.simulator_proto_unix_io.send_proto(SimulatorTick, tick)
+                time.sleep(tick_rate_ms / 1000)
+
+        # Launch all binaries
+        with Simulator(
+            args.simulator_runtime_dir, args.debug_simulator
+        ) as simulator, FullSystem(
+            args.blue_fullsystem_runtime_dir, args.debug_fullsystem, False
+        ) as blue_fs, FullSystem(
+            args.yellow_fullsystem_runtime_dir, args.debug_fullsystem, True
+        ) as yellow_fs, Gamecontroller() as gamecontroller:
+
+            blue_fs.setup_proto_unix_io(tscope.blue_full_system_proto_unix_io)
+            yellow_fs.setup_proto_unix_io(tscope.yellow_full_system_proto_unix_io)
+            simulator.setup_proto_unix_io(
+                tscope.simulator_proto_unix_io,
+                tscope.blue_full_system_proto_unix_io,
+                tscope.yellow_full_system_proto_unix_io,
+            )
+            gamecontroller.setup_proto_unix_io(
+                tscope.blue_full_system_proto_unix_io,
+                tscope.yellow_full_system_proto_unix_io,
+            )
+
+            tscope.load_saved_layout(args.layout)
+
+            # Start the simulator
+            thread = threading.Thread(
+                target=__async_sim_ticker, args=(SIM_TICK_RATE_MS,), daemon=True,
+            )
+
+            thread.start()
+            tscope.show()
+            thread.join()
+            tscope.close()
```

Comment:
I seem to have a leftover process when I ctrl-c thunderscope

Reply:
discussed offline: non issue